### ISSUE TITLE
Dynamic speed increase on silent parts

### ISSUE THREAD

**Issue by GammelSami** [polywock/globalSpeed#64]
Option to dynamically change playback speed if the media loudness is below a certain threshold to speed up silent parts. 
I hope this is possible in browsers with realtime playback.
This video has a graphical explaination what I mean: https://youtu.be/DQ8orIurGxw?t=148

**Comment by polywock** [comment]
I implemented it in the past, but I wasn't very happy with the results. I used the WebAudio API to capture the audio of the video, and used the analyzer node to get the loudness. The problems I remember... 

1. The createMediaElementSource() could only be used once for a video element. If a website is already capturing the audio, I won't be able to capture it. And if I capture it first, and they try to also capture it, it might cause other issues.  

2. Some video's audio could not be captured depending on the website configuration. I was getting lots of CORS / cross-origin errors. 

3. Because it's a realtime playback, you can't analyze all the sound at once. That means you can't lookahead to see if volume stays below a certain threshold. So often, you often have to lower/increase speed very quickly.  And you also have to analyze the current sound segment every X seconds or so. The more frequently you check, the more accurate it is. But, also the more computational power it costs.   
 
The short version, it's buggy and laggy.

**Comment by GammelSami** [comment]
I just found out that the Google Podcasts App has such a feature. At least a workaround for some media :)
![Google Podcasts Playback Speed Settings German](https://pbs.twimg.com/media/Exa3958XAAQ9qm1?format=jpg&name=medium)
